src/llm_utils.py

```diff
@@ -0,0 +1,190 @@
+
+    Example response:
+    FILE: .github/workflows/review.yml
+    LINE: 12
+    COMMENT: Consider adding branch filters
+    SUGGESTION: |
+    on:
+        pull_request:
+        branches: [main, develop]
+
+    ---
+
+    Diff content:
+    {diff_content}
+    """
+
+    client = genai.Client(api_key=API_KEY)
+
+    #Get analysis from Gemini model
+    response = client.models.generate_content(
+        model="gemini-2.0-flash", contents=prompt
+    )   
+
+    print(response.text)
+    
+    # Parse and format the response
+    review_comments = parse_llm_response(response.text)
+    return review_comments
+
+
+
+"""
+This is the final code given by deepseek to parse the response from the LLM model. It includes errors
+"""
+
+def parse_llm_response(response: str) -> List[Dict]:
+    review_comments = []
+    
+    # Split into comment blocks while preserving code blocks
+    comment_blocks = []
+    current_block = []
+    in_code_block = False
+    
+    for line in response.split('\n'):
+        if line.strip().startswith('---') and not in_code_block:
+            if current_block:
+                comment_blocks.append('\n'.join(current_block))
+                current_block = []
+        elif line.strip().startswith('```'):
+            in_code_block = not in_code_block
+            current_block.append(line)
+        else:
+            current_block.append(line)
+    
+    if current_block:
+        comment_blocks.append('\n'.join(current_block))
+
+    for block in comment_blocks:
+        comment_data = {
+            'path': None,
+            'line': None,
+            'body': []
+        }
+        
+        current_section = None
+        
+        for line in block.split('\n'):
+            line = line.rstrip()
+            
+            if line.startswith('FILE:'):
+                comment_data['path'] = line.split('FILE:', 1)[1].strip()
+            elif line.startswith('LINE:'):
+                try:
+                    comment_data['line'] = int(line.split('LINE:', 1)[1].strip())
+                except (ValueError, IndexError):
+                    continue
+            elif line.startswith('COMMENT:'):
+                comment_data['body'].append(line.split('COMMENT:', 1)[1].strip())
+            elif line.startswith('SUGGESTION:'):
+                suggestion = line.split('SUGGESTION:', 1)[1].strip()
+                comment_data['body'].append(f"\n```suggestion\n{suggestion}\n```")
+            elif line and any([line.startswith(s) for s in ['FILE:', 'LINE:', 'COMMENT:', 'SUGGESTION:']]):
+                continue  # Skip empty lines after headers
+            else:
+                if comment_data['body']:
+                    comment_data['body'].append(line)
+
+        # Validate and format
+        if comment_data['path'] and comment_data['line'] is not None and comment_data['body']:
+            try:
+                formatted_body = '\n'.join(comment_data['body'])
+                # Remove empty suggestion blocks
```

Comment:
The current splitting logic for the comment blocks is complex and may be error-prone. Consider refactoring to improve robustness.

```suggestion
N/A
```